src/main/kotlin/io/emeraldpay/dshackle/rpc/NativeCall.kt

```diff
@@ -101,19 +104,60 @@ open class NativeCall(
     }
 
     open fun nativeCallResult(requestMono: Mono<BlockchainOuterClass.NativeCallRequest>): Flux<CallResult> {
+        val requestSpan = tracer.currentSpan()
         return requestMono.flatMapMany(this::prepareCall)
             .flatMap {
-                if (it.isValid()) {
-                    val parsed = parseParams(it.get())
-                    this.fetch(parsed)
-                        .doOnError { e -> log.warn("Error during native call: ${e.message}") }
+                val requestId = it.requestId
```

Comment:
can we extract this code to separate methods? 
It was small simple one, but now it very big and it hard to understand

Reply:
Yes, extracted this to several methods